### ISSUE TITLE
val logs are interspersed in training logs

### ISSUE THREAD

**Issue by hainanxzg** [ashleve/lightning-hydra-template#164]
2021/08/16 14:12:09 |  
-- | --
2021/08/16 14:12:09 | Validation sanity check: 0it [00:00, ?it/s]
2021/08/16 14:12:09 | Validation sanity check: 0%\| \| 0/2 [00:00<?, ?it/s]
2021/08/16 14:12:09 | Training: -1it [00:00, ?it/s]
2021/08/16 14:12:09 | Training: 0%\| \| 0/74 [00:00<00:00, 49344.75it/s]
2021/08/16 14:12:10 | Epoch 0: 0%\| \| 0/74 [00:00<00:00, 3792.32it/s]
2021/08/16 14:12:10 | Epoch 0: 7%\|##3 \| 5/74 [00:00<00:10, 6.76it/s]
2021/08/16 14:12:11 | Epoch 0: 7%\|8 \| 5/74 [00:00<00:10, 6.76it/s, loss=1.11, v_num=N-35]
2021/08/16 14:12:11 | Epoch 0: 14%\|#4 \| 10/74 [00:01<00:11, 5.51it/s, loss=1.11, v_num=N-35]
2021/08/16 14:12:12 | Epoch 0: 14%\|#4 \| 10/74 [00:01<00:11, 5.50it/s, loss=1.06, v_num=N-35]
2021/08/16 14:12:12 | Epoch 0: 20%\|##2 \| 15/74 [00:02<00:10, 5.51it/s, loss=1.06, v_num=N-35]
2021/08/16 14:12:13 | Epoch 0: 20%\|##2 \| 15/74 [00:02<00:10, 5.50it/s, loss=1.09, v_num=N-35]
2021/08/16 14:12:13 | Epoch 0: 27%\|##9 \| 20/74 [00:03<00:10, 5.28it/s, loss=1.09, v_num=N-35]
2021/08/16 14:12:14 | Epoch 0: 27%\|##9 \| 20/74 [00:03<00:10, 5.28it/s, loss=1.05, v_num=N-35]
2021/08/16 14:12:14 | Epoch 0: 34%\|###7 \| 25/74 [00:04<00:09, 5.36it/s, loss=1.05, v_num=N-35]
2021/08/16 14:12:15 | Epoch 0: 34%\|###3 \| 25/74 [00:04<00:09, 5.36it/s, loss=0.968, v_num=N-35]
2021/08/16 14:12:15 | Epoch 0: 41%\|#### \| 30/74 [00:05<00:08, 5.25it/s, loss=0.968, v_num=N-35]
2021/08/16 14:12:16 | Epoch 0: 41%\|#### \| 30/74 [00:05<00:08, 5.25it/s, loss=0.948, v_num=N-35]
2021/08/16 14:12:16 | Epoch 0: 47%\|####7 \| 35/74 [00:06<00:07, 5.29it/s, loss=0.948, v_num=N-35]
2021/08/16 14:12:17 | Epoch 0: 47%\|####7 \| 35/74 [00:06<00:07, 5.29it/s, loss=0.851, v_num=N-35]
2021/08/16 14:12:17 | Epoch 0: 54%\|#####4 \| 40/74 [00:07<00:06, 5.26it/s, loss=0.851, v_num=N-35]
2021/08/16 14:12:18 | Epoch 0: 54%\|#####4 \| 40/74 [00:07<00:06, 5.25it/s, loss=0.892, v_num=N-35]
2021/08/16 14:12:18 | Epoch 0: 61%\|###### \| 45/74 [00:08<00:05, 5.22it/s, loss=0.892, v_num=N-35]
2021/08/16 14:12:19 | Epoch 0: 61%\|###### \| 45/74 [00:08<00:05, 5.22it/s, loss=0.879, v_num=N-35]
2021/08/16 14:12:19 | Epoch 0: 68%\|######7 \| 50/74 [00:09<00:04, 5.23it/s, loss=0.879, v_num=N-35]
2021/08/16 14:12:20 | Epoch 0: 68%\|#######4 \| 50/74 [00:09<00:04, 5.23it/s, loss=0.83, v_num=N-35]
2021/08/16 14:12:20 | Epoch 0: 74%\|########1 \| 55/74 [00:10<00:03, 5.19it/s, loss=0.83, v_num=N-35]
2021/08/16 14:12:20 | Epoch 0: 74%\|#######4 \| 55/74 [00:10<00:03, 5.19it/s, loss=0.847, v_num=N-35]
2021/08/16 14:12:20 | Epoch 0: 81%\|########1 \| 60/74 [00:11<00:02, 5.31it/s, loss=0.847, v_num=N-35]
2021/08/16 14:12:21 | Validating: 0it [00:00, ?it/s]
2021/08/16 14:12:21 | Validating: 0%\| \| 0/15 [00:00<?, ?it/s]
2021/08/16 14:12:21 | Validating: 33%\|##########6 \| 5/15 [00:00<00:00, 36.64it/s]
2021/08/16 14:12:21 | Epoch 0: 88%\|########7 \| 65/74 [00:11<00:01, 5.68it/s, loss=0.847, v_num=N-35]
2021/08/16 14:12:21 | Validating: 67%\|####################6 \| 10/15 [00:00<00:00, 37.37it/s]
2021/08/16 14:12:21 | Epoch 0: 95%\|#########4\| 70/74 [00:11<00:00, 6.05it/s, loss=0.847, v_num=N-35]
2021/08/16 14:12:23 | Validating: 100%\|###############################\| 15/15 [00:00<00:00, 37.81it/s]
2021/08/16 14:12:23 | Epoch 0: 100%\|#\| 74/74 [00:14<00:00, 5.24it/s, loss=0.827, v_num=N-35, val/loss
2021/08/16 14:12:24 | Epoch 0: 0%\| \| 0/74 [00:00<00:00, 10565.00it/s, loss=0.827, v_num=N-35, val/lo
2021/08/16 14:12:25 | Epoch 1: 0%\| \| 0/74 [00:00<00:00, 474.79it/s, loss=0.827, v_num=N-35, val/loss
2021/08/16 14:12:25 | Epoch 1: 7%\| \| 5/74 [00:00<00:11, 6.04it/s, loss=0.827, v_num=N-35, val/loss=
2021/08/16 14:12:26 | Epoch 1: 7%\| \| 5/74 [00:00<00:11, 6.04it/s, loss=0.828, v_num=N-35, val/loss=
2021/08/16 14:12:26 | Epoch 1: 14%\|1\| 10/74 [00:02<00:12, 5.13it/s, loss=0.828, v_num=N-35, val/loss
2021/08/16 14:12:27 | Epoch 1: 14%\|1\| 10/74 [00:02<00:12, 5.13it/s, loss=0.843, v_num=N-35, val/loss
2021/08/16 14:12:27 | Epoch 1: 20%\|2\| 15/74 [00:03<00:11, 5.26it/s, loss=0.843, v_num=N-35, val/loss
2021/08/16 14:12:28 | Epoch 1: 20%\|2\| 15/74 [00:03<00:11, 5.25it/s, loss=0.885, v_num=N-35, val/loss

**Comment by hainanxzg** [comment]
Is that mean my validation starts before the end of one epoch? But I have already set log_every_n_steps=1.

**Comment by ashleve** [comment]
@hainanxzg in pytorch lightning, full progress bar encapsulates both training epoch and validation epoch (validation is always executed only after training epoch ends, so don't worry about that).

I believe `log_every_n_steps` controls only how often loggers log data during training epoch (e.g. `log_every_n_steps=2` will skip logging metrics for half of the training steps of the epoch).

**Comment by hainanxzg** [comment]
@ashleve thanks, I get it.
